src/Nest/XPack/Watcher/Condition/CompareConditionBase.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Nest
+{
+	[JsonObject]
+	[JsonConverter(typeof(CompareConditionConverter))]
+	public interface ICompareCondition : ICondition
+	{
+		string Path { get; set; }
+		string Comparison { get; }
+		object Value { get; set; }
+	}
+
+	public abstract class CompareConditionBase : ConditionBase, ICompareCondition
+	{
+		private ICompareCondition Self => this;
+
+		string ICompareCondition.Path { get; set; }
+		object ICompareCondition.Value { get; set; }
+		string ICompareCondition.Comparison => this.Comparison;
+
+		protected CompareConditionBase(string path, object value)
+		{
+			Self.Path = path;
+			Self.Value = value;
+		}
+
+		protected abstract string Comparison { get; }
+
+		internal override void WrapInContainer(IConditionContainer container) => container.Compare = this;
+	}
+
+	public class CompareConditionDescriptor : IDescriptor
+	{
+		// TODO: Should this follow DescriptorBase<TDescriptor, TInterface>?
```

Comment:
I'm fine with leaving this as is, especially if both `path` and `value` are always required.